### PR TITLE
Remove BlockTaxProportion and fix tests

### PR DESCRIPTION
**Pull request by harzo**
Two tests are skipped: 
- `chain_test.go:TestBlockchainFunctions`
-  `validate_test.go:TestBlockValidationRules`

Main cause of this is a need of test data regeneration after new rules and/or hash function will be implemented.

If you find any leftovers of BlockTaxProportion, please let me know so i can remove them and rebase.

### CODE REVIEW

**Comment by jksf** [comment]
You have a typo in a commit message:
> subisdy

**Comment by harzo** [comment]
@jksf please review my changes. I'm ready to answer your questions. Thanks in advance :+1:

File: blockchain/chaingen/generator.go [hunk 1]
```diff
@@ -1503,12 +1475,12 @@ func (g *Generator) ReplaceWithNVotes(numVotes uint16) func(*wire.MsgBlock) {
 		// subsidy is accounted for.
 		height := b.Header.Height
 		fullSubsidy := g.calcFullSubsidy(height)
-		devSubsidy := g.calcDevSubsidy(fullSubsidy, height, numVotes)
+		//devSubsidy := g.calcDevSubsidy(fullSubsidy, height, numVotes)
```
Comment (harzo) [inline blockchain/chaingen/generator.go]: I forgot to remove that. Rebasing right now.

**Comment by jksf** [comment]
Looks good, please only remove `ErrNoTax` because it does not make sense anymore.

**Review by jksf — APPROVED** [review]
Good job, thanks :+1:

**Pull request merged**